### ISSUE TITLE
Build warning 'Failed to index boolean ...' in Quarkus < 3.7

### ISSUE THREAD

**Issue by mschorsch** [quarkusio/quarkus#38520]
### Describe the bug

Quarkus < 3.7 shows the following build warning if MediaType.APPLICATION_JSON and return value Boolean. The MediaType is wrong but the error message is not very helpful to identify the problem. 

Note: The problem is not reproducible in Quarkus >= 3.7.

```
[WARNING] [io.quarkus.deployment.steps.CombinedIndexBuildStep] Failed to index boolean: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD for code-with-quarkus-1.0.0-SNAPSHOT@33ff6a62
```

### How to Reproduce?

[reproducer.zip](https://github.com/quarkusio/quarkus/files/14123275/reproducer.zip)

`./mvnw clean install`

### Output of `java -version`

Java 17

### Quarkus version or git rev

3.6.9

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven

**Comment by geoand** [comment]
This was likely fixed in #37456. @Ladicek I believe we can backport it, right?

**Comment by Ladicek** [comment]
Yeah, I think so.

**Comment by gsmet** [comment]
I don't plan on doing any other 3.6 so if it's fixed in 3.7, it's all good (that's what I secretly hoped when I asked for this issue to be opened).

Not sure we would want to backport it to 3.2 (and if it affects 3.2).

**Comment by geoand** [comment]
> I don't plan on doing any other 3.6 

Makes sense!
